go/libraries/doltcore/sqle/dolt_map_iter.go

```diff
@@ -162,7 +162,8 @@ func (conv *KVToSqlRowConverter) processTuple(cols []interface{}, valsToFill int
 		}
 
 		tag64 := primReader.ReadUint()
-		if tag64 > maxTag {
+
+		if tag64 > maxTag && tag64 != schema.KeylessRowCardinalityTag && tag64 != schema.KeylessRowIdTag {
```

Comment:
Keyless index lookups get called with the other path, which sets the largest tag in the column values. So this fails to account for keyless tags that are bigger.  This is also true for your suggested change in index_row_iter.go: `if inPK || (tag == schema.KeylessRowIdTag) {`. Row lookups silently fail in a confusing way without these extra checks. An alternative solution would refactor upstream logic to choose better defaults for keyless rows, but this felt like a smaller change when i was walking through the code. Open to other suggestions, but much of the function stack leading up to row reads make assumptions that aren't true for keyless and I was trying to avoid a bunch of `if sch,IsKeyless { callNewSpecialFunc }`